This patch sets up the default directories to install to
the target the MBR of the first hard disk from C:\factory.

It's analogous to calling grub-setup.exe like this:

# grub-setup -d C:\factory -r "(//./PHYSICALDRIVE0/)" "//./PHYSICALDRIVE0/"

After this patch, it's possible to just install like this:

# grub-setup.exe

Index: grub2-2.02~beta2/util/grub-setup.c
===================================================================
--- grub2-2.02~beta2.orig/util/grub-setup.c	2014-03-11 19:24:15.892737227 +0000
+++ grub2-2.02~beta2/util/grub-setup.c	2014-03-11 19:24:15.884737226 +0000
@@ -207,9 +207,11 @@
         break;
 
       case ARGP_KEY_NO_ARGS:
-          fprintf (stderr, "%s", _("No device is specified.\n"));
-          argp_usage (state);
+          fprintf(stderr, "Installing GRUB2 to `%s'\n", DELL_TARGET);
+          /*
+	  argp_usage (state);
 	  exit (1);
+          */
           break;
 
       default:
@@ -286,7 +288,10 @@
   grub_mdraid1x_init ();
   grub_lvm_init ();
 
-  dest_dev = get_device_name (arguments.device);
+  if (arguments.device)
+      dest_dev = get_device_name (arguments.device);
+  else
+      dest_dev = grub_util_get_grub_dev (DELL_TARGET);
   if (! dest_dev)
     {
       /* Possibly, the user specified an OS device file.  */
@@ -329,5 +334,6 @@
   free (root_dev);
   free (dest_dev);
 
+  fprintf (stderr, "%s", _("Completed successfully.\n"));
   return 0;
 }
Index: grub2-2.02~beta2/include/grub/osdep/hostfile_windows.h
===================================================================
--- grub2-2.02~beta2.orig/include/grub/osdep/hostfile_windows.h	2014-03-11 19:24:15.892737227 +0000
+++ grub2-2.02~beta2/include/grub/osdep/hostfile_windows.h	2014-03-11 19:29:25.852735843 +0000
@@ -28,7 +28,8 @@
 #define GRUB_UTIL_FD_IS_VALID(x) ((x) != GRUB_UTIL_FD_INVALID)
 #define GRUB_UTIL_FD_STAT_IS_FUNCTIONAL 0
 
-#define DEFAULT_DIRECTORY	"C:\\"GRUB_BOOT_DIR_NAME"\\"GRUB_DIR_NAME
+#define DELL_TARGET "//./PHYSICALDRIVE0/"
+#define DEFAULT_DIRECTORY	"C:/factory"
 #define DEFAULT_DEVICE_MAP	DEFAULT_DIRECTORY "/device.map"
 
 struct grub_util_fd_dirent
Index: grub2-2.02~beta2/util/setup.c
===================================================================
--- grub2-2.02~beta2.orig/util/setup.c	2014-03-11 19:24:15.892737227 +0000
+++ grub2-2.02~beta2/util/setup.c	2014-03-11 19:29:19.696736115 +0000
@@ -255,7 +255,7 @@
 {
   char *core_path;
   char *boot_img, *core_img, *boot_path;
-  char *root = 0;
+  char *root = grub_util_get_grub_dev (DELL_TARGET);
   size_t boot_size, core_size;
 #ifdef GRUB_SETUP_BIOS
   grub_uint16_t core_sectors;
@@ -302,13 +302,19 @@
 						   - sizeof (*bl.block));
   grub_util_info ("root is `%s', dest is `%s'", root, dest);
 
+  /* Open the root device and the destination device.  */
+  grub_util_info ("Opening root");
+  root_dev = grub_device_open (root);
+  if (! root_dev)
+    grub_util_error ("%s", grub_errmsg);
+
   grub_util_info ("Opening dest");
   dest_dev = grub_device_open (dest);
   if (! dest_dev)
     grub_util_error ("%s", grub_errmsg);
 
   core_dev = dest_dev;
-
+/*
   {
     char **root_devices = grub_guess_root_devices (dir);
     char **cur;
@@ -353,7 +359,7 @@
     grub_util_info ("guessed root_dev `%s' from "
 		    "dir `%s'", root_dev->disk->name, dir);
   }
-
+*/
   grub_util_info ("setting the root device to `%s'", root);
   if (grub_env_set ("root", root) != GRUB_ERR_NONE)
     grub_util_error ("%s", grub_errmsg);
@@ -524,6 +530,13 @@
 
     assert (nsec <= maxsec);
 
+    /* Ignore failures in the embedding test */
+    if (grub_errno != GRUB_ERR_NONE)
+    {
+        grub_errno = GRUB_ERR_NONE;
+        grub_util_warn("The embedding test declared a failure.  This is normally non-fatal during WinPE, so it will be ignored.");
+    }
+
     /* Clean out the blocklists.  */
     bl.block = bl.first_block;
     while (bl.block->len)
